### PR TITLE
8276129: PretouchTask should page-align the chunk size

### PR DESCRIPTION
**Pull request by kimbarrett**
Please review this change to PretouchTask to page-align the chunk size.
Aligning down is used because it's easy; aligning up would require dealing
with possible overflow or verifying that overflow isn't possible.

Testing:
mach5 tier1

Locally (linux-x64) java -version with pretouch task logging and various
chunk sizes, manually verifying the number of workers was as expected.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8276129](https://bugs.openjdk.java.net/browse/JDK-8276129): PretouchTask should page-align the chunk size

### Reviewers
 * [Stefan Karlsson](https://openjdk.java.net/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to ce8ad82bf9849480fb5f93801cdfe24d0340e080
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to ee12018b9e08b7d307cbef71c205b750dc0e8d3b

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6204/head:pull/6204` \
`$ git checkout pull/6204`

Update a local copy of the PR: \
`$ git checkout pull/6204` \
`$ git pull https://git.openjdk.java.net/jdk pull/6204/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6204`

View PR using the GUI difftool: \
`$ git pr show -t 6204`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6204.diff">https://git.openjdk.java.net/jdk/pull/6204.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/shared/pretouchTask.cpp [hunk 1]
```diff
@@ -67,6 +68,9 @@ void PretouchTask::pretouch(const char* task_name, char* start_address, char* en
   // Chunk size should be at least (unmodified) page size as using multiple threads
   // pretouch on a single page can decrease performance.
   size_t chunk_size = MAX2(PretouchTask::chunk_size(), page_size);
+  // Page-align the chunk size, so if start_address is also page-aligned (as
+  // is common) then there won't be any pages shared by multiple chunks.
+  chunk_size = align_down(chunk_size, page_size);
```
Comment (stefank) [inline src/hotspot/share/gc/shared/pretouchTask.cpp:73]: This looks like a place where align_down_bounded could be used. Something like this:
size_t chunk_size = align_down_bounded(PretouchTask::chunk_size(), page_size);

**Review by stefank — APPROVED** [review]

**Review by tschatzl — APPROVED** [review]
Lgtm independent of whether you want to incorporate @stefank 's suggestion.

File: src/hotspot/share/gc/shared/pretouchTask.cpp (hunk 1 above)
Comment (kimbarrett) [reply src/hotspot/share/gc/shared/pretouchTask.cpp]: I either forgot about or never noticed that function.  Changed to use it.

**Review by stefank — APPROVED** [review]

**Comment by kimbarrett** [comment]
Thanks @stefank and @tschatzl for reviews.

**Pull request closed without merging**